clustered/server/src/test/java/org/ehcache/clustered/server/store/ChainBuilder.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.ehcache.clustered.server.store;
+
+import org.ehcache.clustered.common.store.Chain;
+import org.ehcache.clustered.common.store.Element;
+
+/**
+ * Builds {@link Chain}s
+ */
+public interface ChainBuilder {
```

Comment:
Both the builders were added with intention to enable client side creation of Chains and Elements. But I am not very sure about how they will be used until we figure out the ServerStoreProxy and its impl. FOr now Ihave just moved them to test package and it surely helps in implementing a ServerStore test which does not depend on the reference implementation